src/plugins/intel_cpu/src/nodes/common/cpu_convert.cpp

```diff
@@ -464,14 +599,141 @@ struct ConvertPrecision<std::tuple<ov::float16, ov::float16>> {
                 const size_t current_batch_size = std::min(ctx.size - offset, batch);
                 jit_convert(src + offset, tmp, current_batch_size);     // fp16 -> fp32
                 for (size_t j = 0; j < current_batch_size; ++j)         // truncate fp32
-                    tmp[j] = std::trunc(std::max(std::min(tmp[j], ubound), lbound));
+                    tmp[j] = clamp ? std::trunc(std::max(std::min(tmp[j], ubound), lbound)) : tmp[j];
                 jit_convert(tmp, dst + offset, current_batch_size);     // fp32 -> fp16
             });
         }
 
         ctx.converted = true;
     }
 };
+
+template <>
+struct ConvertPrecision<std::tuple<float, ov::float16>> {
+    void operator()(ConvertContext& ctx) {
+        auto src = static_cast<const float*>(ctx.srcPtr);
+        auto dst = static_cast<ov::float16*>(ctx.dstPtr);
+
+        constexpr size_t batch = 64;
+        const size_t iterations = ov::intel_cpu::div_up(ctx.size, batch);
+        auto clamp = ctx.clamp;
+
+        if (!clamp) {
```

Comment:
Can we unify the if...else... code blocks here by using
`jit_convert<float, ov::float16, clamp>(src + offset, dst + offset, current_batch_size);`

Reply:
combine these two `jit_convert` functions seems will make its usage quite confusing/complicated in `ConvertPrecision` struct